### PR TITLE
Update Watchlist feature initialization

### PR DESCRIPTION
**Pull request by staykids**
**Phabricator:** https://phabricator.wikimedia.org/T335606 (partially)

### Notes
Updates `WKAppEnvironment` theme propagation and the initialization of the Watchlist feature. Depends on merge of https://github.com/wikimedia/wikipedia-ios-components/pull/24.

### Test Steps
1. Use staging build or update `FeatureFlags` to enable Watchlist
2. Add items to your Watchlist
3. Open Watchlist
4. Confirm the Watchlist renders
5. Catch theme, return to Watchlist and confirm new theme is rendered

### CODE REVIEW

**Comment by staykids** [comment]
Note: will need a Components revision update and a `PresentationConfiguration` update before merge.

File: WMF Framework/CommonStrings.swift [hunk 1]
```diff
@@ -276,6 +276,10 @@ public class CommonStrings: NSObject {
     public static let watchlist = WMFLocalizedString("watchlist", value: "Watchlist", comment: "Title for watchlist feature.")
 
     public static let watchlistFilter = WMFLocalizedString("watchlist-filter", value: "Filter", comment: "Title for filter button in watchlist.")
+
+    public static let watchlistByte = WMFLocalizedString("watchlist-byte", value: "byte", comment: "Displayed in watchlist view to indicate a single or no byte change in a revision.")
```
Comment (tonisevener) [inline WMF Framework/CommonStrings.swift:280]: @staykids I think you can remove these strings now.

Comment (staykids) [reply WMF Framework/CommonStrings.swift]: Removed - thanks for catching this!

**Comment by staykids** [comment]
Resolved merge conflicts and updated `Components`/`WKData` revisions and ready for another look!

I noticed some odd behavior surrounding the navigation bar when dismissing the view (related to work in https://github.com/wikimedia/wikipedia-ios-components/pull/22) but lets see how that irons out after remaining features are integrated.

**Review by tonisevener — APPROVED** [review]

**Pull request merged**